core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1910,11 +1910,15 @@ private boolean needNewSubQuery(
         return true;
       }
 
+      // Cannot merge two window functions
+      boolean containsOver = containsOver(node);
       if (rel instanceof Project
           && ((Project) rel).containsOver()
-          && maxClause == Clause.SELECT) {
-        // Cannot merge a Project that contains windowed functions onto an
```

Comment:
you lost this comment. wasn't it useful?

Reply:
Previously, only the merging of Window functions in Project was handled here, but now window functions are handled whether they are in Project or not. So this comment doesn't quite fit anymore.